odd-collector/odd_collector/adapters/ckan/client.py

```diff
@@ -47,16 +52,51 @@ async def _post_request(self, url: str, payload: dict = None) -> dict:
         ) as session:
             try:
                 async with session.post(url, json=payload) as resp:
-                    result = await resp.json()
-                    logger.debug(f"Result of request {url} is {result}")
+                    if resp.status != 200:
+                        logger.debug(
+                            f"POST request for {url} with payload={payload} has not been successful "
+                            f"with a response status code = {resp.status}"
+                        )
+                        result = {"success": False}
+                    else:
+                        result = await resp.json()
+                        logger.debug(
+                            f"POST request for {url} with payload={payload} has been successful "
+                            f"with a result = {result}"
+                        )
                 return result
             except Exception as e:
                 raise DataSourceError(
                     f"Error during getting data from host {self.__host}: {e}"
                 ) from e
 
+    async def _pagination_request(self, url: str, params: dict) -> list:
+        """
+        Custom pagination function for getting all the datasets from request on specified url
+        with given query parameters.
+        params["rows"] and params["start"] are essential for pagination, as "start" specifies
+        the start index of result response and "rows" specifies how much entites we will get
+        once per request.
+        """
+        pagination_interval = params["rows"]
+
+        extra_quit_count = 0
+        result = []
+        while True:
+            extra_quit_count += 1
+
+            resp = await self._get_request(url, params)
+            result.extend(resp["result"]["results"])
+            if resp["result"]["count"] == len(result):
+                break
+            params["start"] += pagination_interval
+
+            if extra_quit_count == 200:
+                break
+        return result
```

Comment:
According the break condition, there is iteration count limit so `while True` construct is misguiding a little bit. Can we do something like this?

pseudo code. I didn't test it.

```suggestion
        pagination_interval = params["rows"]
        result = []

        for _ in range(200):  # Limit the number of iterations to 200
            resp = await self._get_request(url, params)
            result.extend(resp["result"]["results"])

            if resp["result"]["count"] == len(result):
                break

            params["start"] += pagination_interval
        return result
```

Reply:
done